### DISCUSSION TITLE
Orchestrating AWS Glue Jobs with Dagster

### DISCUSSION THREAD

**Discussion by OMEGAJZ** [dagster-io/dagster#19369] [Q&A] [answered]
Hi everyone,

my organization has an AWS Glue spark notebook job which extracts and loads multiple tables into a databse. 

We want to orchestrate our whole ETL process (AWS Glue, dbt) within Dagster / Dagster UI. 

How do we make our AWS Glue job visible and orchestrateable in Dagster and connect it to our dbt transformations?

Thanks in advance!

**Comment by OMEGAJZ** [comment]
My current solution is, that I create an "@op" which loads a boto3 glue client. With boto3 I'm able to access "get_job_runs" and "start_job_run" to kick my glue jobs off.   A "@job" decorater acts as a trigger for my "@op" action. 

Now a schedule is starting my aws glue job and a sensor is tracking the glue job status. 

next steps: I'm going to connect my dbt models to this aws glue/Dagster pipeline.

**Maybe someone did all of this before and can share his insights and solution.** 

thanks!

**Comment by rexledesma** [comment, marked as answer]
We'd suggest you to:

1. Follow the dbt tutorial to get set up with Dagster and dbt: https://docs.dagster.io/integrations/dbt/using-dbt-with-dagster
2. Create a [`@multi_asset`](https://docs.dagster.io/concepts/assets/multi-assets) that materializes your tables by triggering your glue job.
3. Ensure that the outputs of your `@multi_asset` are the dbt sources that you've defined in your dbt project. See our documentation on defining dbt sources as upstream dependencies: https://docs.dagster.io/integrations/dbt/reference#upstream-dependencies
